src/msix/UnicodeConversion.cpp

```diff
@@ -22,10 +23,12 @@ namespace MSIX {
 
     std::wstring utf8_to_wstring(const std::string& utf8string)
     {
+        if (utf8string.empty()) { return {}; }
         #ifdef WIN32
-        int size = MultiByteToWideChar(CP_UTF8, 0, utf8string.data(), utf8string.size(), nullptr, 0);
+        int size = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, utf8string.data(), static_cast<int>(utf8string.size()), nullptr, 0);
+        ThrowLastErrorIf(size == 0, "Error converting to wstring");

```

Comment:
>size [](start = 25, length = 4)

There is an issue here that can be fixed in another change.  This size value includes the null char and thus the returned string does too.

Reply:
MSDN is confusing... first call has cbMultiByte as a positive integer which means: "the function processes exactly the specified number of bytes. If the provided size does not include a terminating null character, the resulting Unicode string is not null-terminated, and the returned length does not include this character". So it won't include the null char.

---
In reply to: [289088558](https://github.com/Microsoft/msix-packaging/pull/149#discussion_r289088558) [](ancestors = 289088558)